### PR TITLE
8269850: Most JDK releases report macOS version 12 as 10.16 instead of 12.0

### PR DESCRIPTION
**Pull request by RogerRiggs**
The Mac OS specific code to determine the os.version property in java_props_macosx.c is updated
to replace the code extracting the version from the SystemVersion.plist by reading the version using t\
he hidden link:
/System/Library/CoreServices/.SystemVersionPlatform.plist

Its contents are not dependent on the SYSTEM_VERSION_COMPAT environment variable.
With this change, 11.5 reports the 11.5.1 minor version and os.version 11.6 is correctly reported.
The change can be backported to other versions as needed.

(Opening a new PR, the PR branch #5619 was deleted by unintentionally).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269850](https://bugs.openjdk.java.net/browse/JDK-8269850): Most JDK releases report macOS version 12 as 10.16 instead of 12.0

### Reviewers
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5633/head:pull/5633` \
`$ git checkout pull/5633`

Update a local copy of the PR: \
`$ git checkout pull/5633` \
`$ git pull https://git.openjdk.java.net/jdk pull/5633/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5633`

View PR using the GUI difftool: \
`$ git pr show -t 5633`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5633.diff">https://git.openjdk.java.net/jdk/pull/5633.diff</a>

</details>

### CODE REVIEW

File: src/java.base/macosx/native/libjava/java_props_macosx.c [hunk 1]
```diff
@@ -262,18 +262,13 @@ void setOSNameAndVersion(java_props_t *sprops) {
...
             if (version != NULL) {
-                NSString *nsBuildVerStr = [version objectForKey : @"ProductBuildVersion"];
-                if (nsBuildVerStr != NULL && nsBuildVerStr.length >= 3) {
-                    int letter = [nsBuildVerStr characterAtIndex:2];
-                    if (letter >= 'B' && letter <= 'Z') {
-                        int vers = letter - 'A' - 1;
-                        asprintf(&osVersionCStr, "11.%d", vers);
-                    }
-                }
+                NSString *nsVerStr = [version objectForKey : @"ProductVersion"];
```
Comment (naotoj) [inline src/java.base/macosx/native/libjava/java_props_macosx.c:270]: Should we check `nsVerStr != NULL` here?

**Review by naotoj — APPROVED** [review]

**Review by RealCLanger — APPROVED** [review]
Looks good to me. I tested with an Xcode 11.7 build on a MacOS 11.6 and there OsVersionTest fails without the fix and passes with it.

**Comment by RogerRiggs** [comment]
Thanks for testing with Xcode 11.7. and MacOS 11.6.

**Pull request closed without merging**